mne/source_space.py

```diff
@@ -3197,3 +3199,90 @@ def _get_src_nn(s, use_cps=True, vertices=None):
     else:
         nn = s['nn'][vertices, :]
     return nn
+
+
+@verbose
+def vertex_depths(src, info=None, picks=None, trans=None, mode=None,
```

Comment:
I would keep the API simple and flexible I would have returned an array which is of shape (n_vertices, n_channels) where n_channels is all channels present in info (making it mandatory) or limited to picks is not None. Trans should be required if MEG channels are present. Then mode could be removed from API. I think it's simple enough to do `np.mean(dist, axis=1)`

also maybe rename to compute_src_channel_dist ?

thoughts @olafhauk @larsoner ?

Reply:
> Trans should be required if MEG channels are present.

`trans` will be necessary if `src` is in MRI space (i.e., not from forward or inv), so I don't think it can go away. It's also very common to have to provide `trans` when providing a `src` so I think it's good to keep it.

> I would have returned an array which is of shape (n_vertices, n_channels)

Sounds good to me, indeed it's easy enough to post-process this to get what you need. And generally `n_channels` is an order of magnitude smaller than `n_vertices` so it shouldn't be too slow or memory hungry.

Reply:
Happy to return all distances for post-processing.
vertex_depth() currently only takes src (not fwd). Let me know if you want me to change this.